investd/sources/bonds.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+from typing import Iterator
+
+import pandas as pd
+
+from .base import SourceBase
+from ..model import Action, AssetType, Currency, Transaction
+
+BOND_VALUE = 100
+
+COL_DATE = "DATA DYSPOZYCJI"
+COL_CODE = "KOD OBLIGACJI"
+COL_STATUS = "STATUS"
+COL_TYPE = "RODZAJ DYSPOZYCJI"
+COL_COUNT = "LICZBA OBLIGACJI"
+COL_AMOUNT = "KWOTA OPERACJI"
+
+STATUS_COMPLETED = "zrealizowana"
+TYPE_BUY = "zakup papierów"
+TYPE_PREMATURE_SELL = "przedterminowy wykup"
+
+
+class Bonds(SourceBase):
+    source_name = "bonds"
+
+    def parse_source_file(self, path: Path) -> Iterator[Transaction]:
+        df = pd.read_excel(
+            path,
+            usecols=[COL_DATE, COL_TYPE, COL_CODE, COL_COUNT, COL_AMOUNT, COL_STATUS],
+            parse_dates=[COL_DATE],
+        )
+        df = df[df[COL_STATUS] == STATUS_COMPLETED]
+
+        # correct the number and amount of premature sell operations
+        for i in df.index[df[COL_TYPE] == TYPE_PREMATURE_SELL].tolist():
+            df.at[i, COL_COUNT] = df.at[i, COL_AMOUNT] // BOND_VALUE
+            df.at[i, COL_AMOUNT] = (
+                df.at[i, COL_AMOUNT]
+                + df.at[i + 1, COL_AMOUNT]
+                + df.at[i + 2, COL_AMOUNT]
+            )
```

Comment:
This is interesting. I wonder why it' listed like that in the file. Pretty cool `for statement` though.